### PR TITLE
Reduced and simplified sample native modules

### PR DESCRIPTION
**Pull request by jonthysell**
* Cleaned up SampleApps solution
* SampleLibraryCS now provides SampleModuleCS native module via ReactPackageProvider
* SampleLibraryCPP now provides SampleModuleCPP native module via ReactPackageProvider
* SampleAppCPP and SampleAppCS both provide a DebugConsole native module for logging
* Removed the INativeModule stack

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/3426)

### CODE REVIEW

**Comment by jonthysell** [comment]
This is still in progress, just wanted to have the diff up for comment.

File: packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleApp.vcxproj [hunk 1]
```diff
@@ -167,33 +167,7 @@
   <ItemGroup>
     <Natvis Include="$(ReactNativeWindowsDir)\Folly\Folly.natvis" />
   </ItemGroup>
-  <!-- We need to reference Microsoft.ReactNative to build, but we can't because SampleLibraryCPP already references it, and CppWinRT
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleAppCPP/SampleApp.vcxproj]: ><!-- We ne [](start = 2, length = 10)

I am just curious: was this code the reason why I had all the winmd merge issues? :)

File: packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCPP/ReactPackageProvider.cpp [hunk 2]
```diff
@@ -2,17 +2,19 @@
 // Licensed under the MIT License.
 
 #include "pch.h"
-#include "SampleLibraryCppPackage.h"
-#include "SampleLibraryCppPackage.g.cpp"
-#include "Calculator.h"
-#include "MyModule.h"
+#include "ReactPackageProvider.h"
+#if __has_include("ReactPackageProvider.g.cpp")
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCPP/ReactPackageProvider.cpp]: >__has_include [](start = 4, length = 13)

Why is it needed?

File: packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCPP/SampleModuleCPP.h [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+namespace SampleLibraryCPP {
+
+// Sample REACT_MODULE
+
+REACT_MODULE(SampleModuleCPP);
+struct SampleModuleCPP {
+  const std::string Name = "SampleModuleCPP";
+
+  void DebugWriteLine(const std::string &methodName, const std::string &arg) {
+    std::string output = Name;
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCPP/SampleModuleCPP.h]: >std::string [](start = 4, length = 11)

nit: it is better to use stringstream.

File: packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCS/SampleModuleCS.cs [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+using Microsoft.ReactNative.Managed;
+
+namespace SampleLibraryCS
+{
+    // Sample ReactModule
+
+    [ReactModule]
+    internal sealed class SampleModuleCS
+    {
+        public string Name => nameof(SampleModuleCS);
```
Comment (vmoroz) [inline packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCS/SampleModuleCS.cs]: >public string Name => nameof(SampleModuleCS); [](start = 8, length = 45)

It must not be needed.

Comment (vmoroz) [reply packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCS/SampleModuleCS.cs]: The ReactNoduleAttribute by default returns name of the class.

---
In reply to: [335169016](https://github.com/microsoft/react-native-windows/pull/3426#discussion_r335169016) [](ancestors = 335169016)

Comment (vmoroz) [reply packages/microsoft-reactnative-sampleapps/windows/SampleLibraryCS/SampleModuleCS.cs]: Please ignore it - I see it is being used. Maybe it should private then.

---
In reply to: [335169247](https://github.com/microsoft/react-native-windows/pull/3426#discussion_r335169247) [](ancestors = 335169247,335169016)

**Pull request merged**